src/persist-client/src/impl/machine.rs

```diff
@@ -316,6 +316,17 @@ where
                         new_state
                     );
                     self.state = new_state;
+
+                    // Bound the number of entries in consensus.
+                    //
+                    // It's weird that this will return an error for the whole
+                    // request after we know it's successful, but this goes away
+                    // in #12223.
+                    let () = self
```

Comment:
We're doubling (yes, 2 😅) the number of consensus operations with this. We could maybe get around it with a combined `compare_and_append_and_truncate()` operation, but I don't know if I like that either.

I think it's fine to merge as is, but still wanted to point it out.

Reply:
yup! we could also amortize it (e.g. only call every log(N) success or something) if that ends up being an issue